packages/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -39,6 +37,7 @@ const navigationNodeToEuiItem = (
     [`nav-item-isActive`]: isSelected,
   });
 
+  // @ts-expect-error getRenderItem is not used (doesn't exist on EuiCollapsibleNavSubItemProps)
```

Comment:
FIXME: Not sure what this was originally doing, but it probably shouldn't just be unused

Reply:
my (contextless) 2c: I'd suggest removing it entirely, since the new component won't use it.

Reply:
@cee-chen 

Seb caught the problem with this removal here: https://github.com/elastic/kibana/pull/164910#discussion_r1314588325

This was providing functionality that allowed teams to render any content inside the navigation node. It's important we're still able to do this: the Security Solution team plans to start using our navigation node types, and include the flyout menus that we're familiar with. To do that, I believe we need [the `renderItem` support that exists in EuiSideNavItemType](https://github.com/elastic/eui/blob/d4561360a99324ce1bc3c65b7f6c00ebdae17df1/src/components/side_nav/side_nav_types.ts#L39-L42)

@sebelga can you confirm this?

Reply:
Haha, we're mindmelding right now. I responded at the exact same time with a comment down below: https://github.com/elastic/kibana/pull/164910#discussion_r1326632602

I think the only thing that's not totally clear to me is whether Kibana needs the ability to custom render just top level items (if so: doesn't need an EUI enhancement), or if it needs the ability to custom render nested items (if so: does need an EUI enhancement).

Reply:
In a separate PR, we will update the API for declaring the navigation tree to allow consumers to declare flyouts. The props for the API change will be handled at a higher level: if a navigation item declares a flyout, a wrapper around our `EuiCollapsibleNavItem` will defer to another way of rendering the secondary action and enabling the flyout. 

Added 1f95d3c9d9d

cc @sebelga